apps/studio/components/interfaces/Connect/Connect.tsx

```diff
@@ -264,6 +266,20 @@ export const Connect = () => {
     }
   }
 
+  useEffect(() => {
+    if (!showConnect) return
+    const hasConnectTabInUrl = typeof router.query.connectTab !== 'undefined'
+    if (!hasConnectTabInUrl && (queryFramework || queryUsing || queryWith)) {
+      const inferred = inferConnectTabFromParentKey(queryFramework)
+      if (inferred) {
+        setTab(inferred)
+        if (inferred === 'frameworks') setConnectionObject(FRAMEWORKS)
+        if (inferred === 'mobiles') setConnectionObject(MOBILES)
+        if (inferred === 'orms') setConnectionObject(ORMS)
+      }
+    }
```

Comment:
Can this if statement be evaluated in the previous clause? Not sure why its nested like this. So something like the following


```suggestion
    const noConnectTabInUrl = typeof router.query.connectTab === 'undefined'
    const hasQuery = queryFramework || queryUsing || queryWith
    const inferred = inferConnectTagFromParentKey(queryFramework)
    if (!hasConnectTabInUrl && hasQuery && inferred) {
        setTab(inferred)
        if (inferred === 'frameworks') setConnectionObject(FRAMEWORKS)
        if (inferred === 'mobiles') setConnectionObject(MOBILES)
        if (inferred === 'orms') setConnectionObject(ORMS)
    }
```